File: mingachevir-real-estate-site-master/src/app/models/request/GetFilterHousesRequest.ts
import {OrderEnum} from "../../util/OrderEnum";
import {InputParameterValueRequest} from "./InputParameterValueRequest";

export interface GetFilterHousesRequest {
  sort?: OrderEnum;
  searchKey?: string;
  mainCategoryId?: number;
  subCategoryId?: number;
  selectiveParameterIds?: Array<number>;
  inputParametersRanges?: Array<InputParameterValueRequest>;
  pageNumber?: number;
}

File: mingachevir-real-estate-server-master/src/main/java/io/mingachevir/mingachevirrealestateserver/model/request/GetFilterHousesRequest.java
package io.mingachevir.mingachevirrealestateserver.model.request;

import io.mingachevir.mingachevirrealestateserver.util.OrderEnum;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.domain.Pageable;

import java.util.List;

@Getter
@Setter
@Data
public class GetFilterHousesRequest {
    private OrderEnum sort;
    private String searchKey;
    private Long mainCategoryId;
    private Long subCategoryId;
    private List<Long> selectiveParameterIds;
    private List<InputParameterValueRequest> inputParametersRanges;
    private Integer pageNumber;
 }
